ui/src/App.tsx

```diff
@@ -178,12 +197,24 @@ export default function App() {
       console.error("app.graph is null cannot load workflow");
       return;
     }
-    setCurFlowID(id);
+
     const flow = getWorkflow(id);
+
     if (flow == null) {
       alert("Error: Workflow not found! id: " + id);
       return;
     }
+
+    if (getWorkflowIdInUrlHash()) {
+      const newUrlHash = generateUrlHashWithFlowId(id);
+      window.location.hash = newUrlHash;
+    } else {
+      localStorage.setItem("curFlowID", id);
+    }
+
+    curFlowID.current = id;
+    setFlowID(id);
```

Comment:
why delete setCurflowID() instead of making this the content of setCurflowID()
We need a unified way to change curFlowID otherwise its easy to get bugs, 
for example these things are always done together:

curFlowID.current = flow.id;
        setFlowID(flow.id);